tests/v2/validation/configmaps/generated_fields_test.go

```diff
@@ -0,0 +1,174 @@
+package configmaps
+
+import (
+	"bytes"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	steveV1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/kubeapi/configmaps"
+	"github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	v1 "k8s.io/api/core/v1"
+	"net/http"
+	"strings"
+	"testing"
+)
+
+const (
+	APIVersion = "v1"
+	Kind       = "ConfigMap"
+)
+
+type ConfigMapTestSuite struct {
+	suite.Suite
+	client           *rancher.Client
+	steveClient      *steveV1.Client
+	session          *session.Session
+	cluster          *management.Cluster
+	httpClient       *http.Client
+	configMapPayload *configMapPayload
+}
+
+type configMapPayload struct {
+	APIVersion string            `yaml:"apiVersion"`
+	Data       map[string]string `yaml:"data"`
+	Kind       string            `yaml:"kind"`
+	Metadata   struct {
+		Fields          string            `yaml:"fields"`
+		Relationships   string            `yaml:"relationships"`
+		State           string            `yaml:"state"`
+		Labels          map[string]string `yaml:"labels"`
+		Name            string            `yaml:"name"`
+		Namespace       string            `yaml:"namespace"`
+		ResourceVersion string            `yaml:"resourceVersion"`
+		UID             string            `yaml:"uid"`
+	} `yaml:"metadata"`
+}
+
+func (c *ConfigMapTestSuite) TearDownSuite() {
+	c.session.Cleanup()
+}
+
+func (c *ConfigMapTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	c.session = testSession
+	client, err := rancher.NewClient("", c.session)
+	require.NoError(c.T(), err)
+	c.client = client
+	c.steveClient = client.Steve
+	c.httpClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+}
+
+func (c *ConfigMapTestSuite) TestSteveGeneratedFields() {
+
+	configData := map[string]string{
+		"env": "qa",
+	}
+	labels, annotations := map[string]string{}, map[string]string{}
+	configMapName := namegenerator.AppendRandomString("test-configmap")
+	configmap, err := configmaps.CreateConfigMap(c.client, c.client.RancherConfig.ClusterName, configMapName, "auto-generated", "default", configData, labels, annotations)
```

Comment:
may be we can add the default namespace as a constant? that way if we want to create additional tests, we can always use that constant instead

Reply:
Added namespace as a constant, also the endpoint "configmaps" as a constant